web/pdf_page_view.js

```diff
@@ -800,10 +809,21 @@ class PDFPageView {
 
     const sfx = approximateFraction(outputScale.sx);
     const sfy = approximateFraction(outputScale.sy);
-    canvas.width = roundToDivide(viewport.width * outputScale.sx, sfx[0]);
-    canvas.height = roundToDivide(viewport.height * outputScale.sy, sfy[0]);
-    canvas.style.width = roundToDivide(viewport.width, sfx[1]) + "px";
-    canvas.style.height = roundToDivide(viewport.height, sfy[1]) + "px";
+
+    const hdpiScales = (w, h) => ({
```

Comment:
Following the previous suggestion, perhaps `const canvasDimensions = ...` here and then do `annotationCanvasDimensions: canvasDimensions,` below at line https://github.com/mozilla/pdf.js/pull/14247/files#diff-c48e3561004f5db8f11d5ebab2fd661591222ba911cb4173fbced15f026bac6bR841